core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -51,15 +52,9 @@ case class ClientSensors(metricTags: Map[String, String], quotaSensor: Sensor, t
  *
  */
 case class ClientQuotaManagerConfig(numQuotaSamples: Int =
```

Comment:
We have new `ClientQuotaManagerConfig.java`.
In the same time scala version of class is here, also.

It seems like we need to keep only one of them.
I think changes related to `ClientQuotaManagerConfig` can be moved to another PR to simplify review and reduce changes.

Reply:
Defaults depends only on `ClientQuotaManagerConfig` object companion only. Moving `ClientQuotaManagerConfig` class can happened when we move `ClientQuotaManager` or in another pr as this will impact QuotaFactory and some Quota test classes

Reply:
In the JIRA we agreed on only moving the object companions

Reply:
Do we really need instance fields in java version in this case?
Maybe it will be more clear to keep only constants.

Reply:
> Do we really need instance fields in java version in this case?
Maybe it will be more clear to keep only constants.

remove the contractor make sense for now

Reply:
#15159 I moved this to another pr